src/java/org/streamingpool/core/conf/DefaultSchedulerConfiguration.java

```diff
@@ -0,0 +1,66 @@
+// @formatter:off
+/*
+*
+* This file is part of streaming pool (http://www.streamingpool.org).
+*
+* Copyright (c) 2017-present, CERN. All rights reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+*/
+// @formatter:on
+
+package org.streamingpool.core.conf;
+
+import static java.util.Arrays.stream;
+import static org.streamingpool.core.conf.TestSchedulerConfiguration.STREAMINGPOOL_TEST_SCHEDULER;
+
+import java.util.concurrent.Executors;
+
+import io.reactivex.Scheduler;
+import io.reactivex.schedulers.Schedulers;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Condition;
+import org.springframework.context.annotation.ConditionContext;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.env.Environment;
+import org.springframework.core.type.AnnotatedTypeMetadata;
+
+@Configuration
+public class DefaultSchedulerConfiguration {
+
+    public static final String STREAMINGPOOL_THREAD_POOL_SIZE = "streamingpool.threadPoolSize";
+
+
+    @Value("${" + STREAMINGPOOL_THREAD_POOL_SIZE + ":100}")
```

Comment:
agree on the system properties in general (we cannot fully escape from system properties since even rxjava uses them for the default buffer size.....). However this defines the behaviour of the default scheduler, if this is not picked up in Spring one can still inject a custom one. Not fully true at the moment since we use it in the main streamingpool configuration, but we could change it :)